keystorage/interlock/src/main/java/tech/pegasys/signers/interlock/handlers/AbstractHandler.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.signers.interlock.handlers;
+
+import tech.pegasys.signers.interlock.InterlockClientException;
+
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import io.vertx.core.MultiMap;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpClientResponse;
+import io.vertx.core.json.JsonObject;
+
+public abstract class AbstractHandler<T> {
```

Comment:
Is Vertx the right choice? Seems like we are doing a lot of work to bend Vertx to behave in a synchronous manner. Perhaps just use the Java http client?

Reply:
I was really tempted to use Apache Http client since it is closeable and all. However, we are using vertx http client in other signers modules (hashicorp for instance). If @jframe and @rain-on are OK with changing to Apache HttpClient, this would be trivial change moving away from vertx.